File: backend/src/models/bill.js
const mongoose = require('mongoose');

const billSchema = new mongoose.Schema(
  {
    savedBillerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'SavedBiller',
      required: true,
    },
    amount: {
      type: Number,
      required: true,
    },
    dueDate: {
      type: Date,
      required: true,
    },
    billingPeriod: {
      type: String,
      required: true,
    },
    isDue: {
      type: Boolean,
      required: true,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Bill', billSchema);

File: lib/features/billers/domain/bill.dart
class Bill {
  final String id;
  final String savedBillerId;
  final double amount;
  final DateTime dueDate;
  final String billingPeriod;
  final bool isDue;

  const Bill({
    required this.id,
    required this.savedBillerId,
    required this.amount,
    required this.dueDate,
    required this.billingPeriod,
    required this.isDue,
  });
}